utils/dataimport.py

```diff
@@ -316,10 +317,18 @@ def import_nordigen(data):
                     raise Exception("Cannot handle different currencies")
                 amount = one["transactionAmount"]["amount"]
                 reference = one.get("entryReference") or ""
+                # clear leading zeroes from real reference
                 reference = reference.lstrip("0")
                 sender = one.get("debtorName", "")
                 message = one.get("additionalInformation", "")
 
+                # If reference is empty, try extracting the first number from the message
+                if not reference:
```

Comment:
Wouldn't it be fairly simple to add this to the holvi function also?

Reply:
the proper way might probably be to do this in the transaction class itself..